SoftLayer/managers/vs.py

```diff
@@ -874,9 +879,30 @@ def upgrade(self, instance_id, cpus=None, memory=None, nic_speed=None, public=Tr
                 raise exceptions.SoftLayerError(
                     "Unable to find %s option with value %s" % (option, value))
 
-            prices.append({'id': price_id})
-        order['prices'] = prices
-
+            if disk is not None:
+                category = {'categories': [{
+                    'categoryCode': 'guest_disk' + str(disk.get('number')),
+                    'complexType': "SoftLayer_Product_Item_Category"
+                }], 'complexType': 'SoftLayer_Product_Item_Price'}
+                prices.append(category)
+                prices[0]['id'] = price_id
```

Comment:
You should set the id in the `category` object. Since its not certain that prices[0] will be the disk price.

```python
                category = {'categories': [{
                    'categoryCode': 'guest_disk' + str(disk.get('number')),
                    'complexType': "SoftLayer_Product_Item_Category"
                }], 'complexType': 'SoftLayer_Product_Item_Price',
                    'id': price_id}
                prices.append(category)
```